### ISSUE TITLE
Entity.StopSound will silence a sound if it's played after the call on the same frame

### ISSUE THREAD

**Issue by Kinyom** [Facepunch/garrysmod-requests#1503]
Sorry for the long title; I'll try to explain this better here:
So basically, I'm writing a SWEP, and I want to silence my various foley/reload sounds when the player fires (CHAN_STATIC is used for firing sounds, as I want them to be able to "overlap" rather than each one cancelling out the last) - to do this, I call use Entity.StopSound calls.

There are some cases, however, where a player will shoot and reload in such quick succession that both happen on the exact same frame; in these scenarios, the StopSound calls are made before the reloading, which then calls one of the reloading sounds that were just previously stopped. When this happens, the reloading sounds will not play - and I'm 100% sure that this isn't the result of PrimaryAttack being called a frame afterwards/on the same frame but a bit later, because if that were the case I'd be firing two bullets; not to mention, if I go with the approach of just stopping CHAN_WEAPON entirely with a silent sound, everything works perfectly.

In short, I think there must be an oversight in the C++ that makes StopSound stay active for an entire tick, silencing the cancelled sound even if it's played after the call is made.

**Comment by robotboy655** [comment]
What's more likely to be the case is that it's just a networking race condition which we can't really do much about.

You can try using `sv_soundemitter_trace 1`, maybe it will shed more light on what is going on.
